Public/Src/Tools/BuildXL.Tools.CredentialProvider/CredentialProviderGenerator.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace BuildXL.Tools.CredentialProvider
+{
+    class CredentialProviderGenerator
+    {
+        private static Dictionary<string, string> ValidPackageSources = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase)
+        {
+            { "https://pkgs.dev.azure.com/1essharedassets/_packaging/BuildXL/nuget/v3/index.json", "1ESSHAREDASSETS_BUILDXL_FEED_PAT" },

```

Comment:
nit: the new syntax looks nicer
```suggestion
            ["https://pkgs.dev.azure.com/1essharedassets/_packaging/BuildXL/nuget/v3/index.json"] = "1ESSHAREDASSETS_BUILDXL_FEED_PAT",
```